src/orders/components/OrderDraftDetailsSummary/OrderDraftDetailsSummary.tsx

```diff
@@ -17,125 +22,204 @@ const useStyles = makeStyles(
     },
     textRight: {
       textAlign: "right"
+    },
+    subtitle: {
+      color: theme.palette.grey[500],
+      paddingRight: theme.spacing(1)
+    },
+    relativeRow: {
+      position: "relative"
+    },
+    percentDiscountLabelContainer: {
+      display: "flex",
+      flexDirection: "row",
+      alignItems: "baseline",
+      justifyContent: "flex-end"
     }
   }),
   { name: "OrderDraftDetailsSummary" }
 );
 
-interface OrderDraftDetailsSummaryProps {
+const messages = defineMessages({
+  subtotal: {
+    defaultMessage: "Subtotal",
+    description: "subtotal price"
+  },
+  addDiscount: {
+    defaultMessage: "Add Discount",
+    description: "add discount button"
+  },
+  discount: {
+    defaultMessage: "Discount",
+    description: "discount button"
+  },
+  addShippingCarrier: {
+    defaultMessage: "Add shipping carrier",
+    description: "add shipping carrier button"
+  },
+  noShippingCarriers: {
+    defaultMessage: "No applicable shipping carriers",
+    description: "no shipping carriers title"
+  },
+  total: {
+    defaultMessage: "Total",
+    description: "total price"
+  },
+  taxes: {
+    defaultMessage: "Taxes (VAT in cluded)",
+    description: "taxes title"
+  }
+});
+
+const PRICE_PLACEHOLDER = "---";
+
+interface OrderDraftDetailsSummaryProps
+  extends OrderDiscountContextConsumerProps {
   disabled?: boolean;
   order: OrderDetails_order;
   onShippingMethodEdit: () => void;
 }
 
 const OrderDraftDetailsSummary: React.FC<OrderDraftDetailsSummaryProps> = props => {
-  const { disabled, order, onShippingMethodEdit } = props;
+  const {
+    order,
+    onShippingMethodEdit,
+    orderDiscount,
+    addOrderDiscount,
+    removeOrderDiscount,
+    openDialog,
+    closeDialog,
+    isDialogOpen,
+    orderDiscountAddStatus,
+    orderDiscountRemoveStatus,
+    undiscountedPrice
+  } = props;
 
+  const intl = useIntl();
   const classes = useStyles(props);
 
+  const popperAnchorRef = useRef<HTMLTableRowElement | null>(null);
+
+  if (!order) {
+    return null;
+  }
+
+  const {
+    subtotal,
+    total,
+    shippingMethod,
+    shippingMethodName,
+    availableShippingMethods,
+    shippingPrice
+  } = order;
+
+  const hasChosenShippingMethod =
+    shippingMethod !== undefined && shippingMethodName !== undefined;
+
+  const hasNoneShippingChosen = shippingMethod === null;
+
+  const hasAvailableShippingMethods = !!availableShippingMethods?.length;
+
+  const discountTitle = !!orderDiscount
```

Comment:
`?` casts the initial value to boolean automatically, no need to do it explicitly